File: 1360-number-of-days-between-two-dates/1360-number-of-days-between-two-dates.py
class Solution:
    def daysBetweenDates(self, date1: str, date2: str) -> int:
        def isLeap(year):
            return year % 400 == 0 or (year % 4 == 0 and year % 100 != 0)

        def daysFrom1971(date):
            year, month, day = map(int, date.split("-"))

            days = 0

            for y in range(1971, year):
                days += 366 if isLeap(y) else 365

            monthDays = [31, 28, 31, 30, 31, 30,
                         31, 31, 30, 31, 30, 31]

            for m in range(month - 1):
                days += monthDays[m]

            if month > 2 and isLeap(year):
                days += 1

            days += day

            return days

        return abs(daysFrom1971(date1) - daysFrom1971(date2))

File: 1360-number-of-days-between-two-dates/1360-number-of-days-between-two-dates.cpp
class Solution {
public:
    bool isLeap(int year) {
        return (year % 400 == 0) || (year % 4 == 0 && year % 100 != 0);
    }

    int daysFrom1971(string date) {
        int year = stoi(date.substr(0, 4));

        static vector<int> daysInMonth = {
            31,28,31,30,31,30,31,31,30,31,30,31
        };

        int days = 0;

        for (int y = 1971; y < year; y++)
            days += isLeap(y) ? 366 : 365;

        for (int m = 1; m < stoi(date.substr(5, 2)); m++) {
            days += daysInMonth[m - 1];
            if (m == 2 && isLeap(year))
                days++;
        }

        days += stoi(date.substr(8, 2));

        return days;
    }

    int daysBetweenDates(string date1, string date2) {
        return abs(daysFrom1971(date1) - daysFrom1971(date2));
    }
};